docs/user-interaction-config-file-spec.rst

```diff
@@ -0,0 +1,204 @@
+
+The ``post_install_tools_disabled`` key corresponds to using ``firstboot --disable`` command in the
+installation kickstart, that requests the post installation setup tools to be skipped. If this key is present
+and set to 1, then any post installation tools that parses the user interaction file should make sure
+it won't be started again and then terminate immediately.
+
+Section headers each correspond to a spoke screen shown in Anaconda. Hubs are not represented as section
+as there is nothing that can be directly set on a hub in Anaconda.
+All currently visible spokes in Anaconda will be enumerated as a section headers by Anaconda in the
+configuration file. Note that this can include spokes provided by addons[1], if any Anaconda addons that provide
+additional spokes are present during the installation run.
+
+
+Section naming
+--------------
+
+Section headers are named according to the Anaconda spoke class name. For example ``DatetimeSpoke``
+or ``KeyboardSpoke``.
+
+To get a list of all such spokes run the ``list_spokes`` script from the ``scripts`` directory in the Anaconda source
+code tree:
+
+::
+
+    git clone https://github.com/rhinstaller/anaconda
+    cd anaconda/scripts
+    ./list_screens
+
+Note that this script only lists Anaconda spokes, not spokes provided by addons[1] or Initial Setup.
+
+It is also possible to check for the *Entered spoke:* entries in the ``/tmp/anaconda.log`` file during an installation
+to correlate spokes on the screen to spoke names.
+
+
+Section namespace
+-----------------
+
+Each section *must* contain the ``visited`` key with a value of either 1 if the user has visited
+the corresponding screen or 0 if not.
+
+Optionally each section can contain one or more keys with the ``changed_`` prefix which track if the user
+has changed an option on the screen. If the option is changed by the user, the corresponding key is set
+to 1. If the given option has not been changed by the user then the corresponding key can either be
+omitted or set to 0.
+
+Example:
+
+::
+
+    [DatetimeSpoke]
+    visited=1
+    changed_timezone=1
+    changed_ntp=0
+    changed_timedate=1
+
+In this example the user has visited the date & time spoke and has changed the timezone & time/date,
+but not the NTP settings. Note that the ``changed_ntp`` key could also be omitted as the user has not changed
+the NTP options.
+
+Another example:
+
+::
+
+    [KeyboardSpoke]
+    visited=0
+
+Here the user has not visited the keyboard spoke and thus could not have changed any options,
+so all ``changed_*`` keys (if any) have been omitted.
+
+Note that if a spoke section is missing, it should be assumed that the corresponding screen has not been visited.
+On othe other hand, if a screen *has been visited*, the section *must* be present, with the ``visited`` key being equal to 1.
+
+
+Full configuration file example
+-------------------------------
+
+::
+
+    post_install_tools_disabled=0
```

Comment:
Matthias points out this is really easy to workaround:

mclasen:  mcatanzaro: you can probably get around the ini-file limbo problem by prepending a group to the data from the file, but getting the format fixes is certainly better
mcatanzaro:  mclasen: I suppose that is a trivial workaround indeed, just add the group, then remove it when we're done with it